clustered/client/src/main/java/org/ehcache/clustered/client/internal/config/xml/ClusteringServiceConfigurationParser.java

```diff
@@ -227,14 +251,197 @@ public URI getNamespace() {
   }
 
+  /**
+   * Translates a {@link ServiceCreationConfiguration} to an xml element
+   *
+   * @param doc
+   *
+   * @param serviceCreationConfiguration
+   */
   @Override
   public void translateServiceConfiguration(Document doc, ServiceCreationConfiguration<ClusteringService> serviceCreationConfiguration) {
+    validateParametersForTranslationToServiceConfig(doc, serviceCreationConfiguration);
+    Element rootElement;
+    try {
+      ClusteringServiceConfiguration clusteringServiceConfiguration = (ClusteringServiceConfiguration)serviceCreationConfiguration;
+
+      rootElement = createConnectionElement(doc, clusteringServiceConfiguration);
+      processTimeUnits(doc, rootElement, clusteringServiceConfiguration);
+      Element serverSideConfigurationElem = processServerSideElements(doc, clusteringServiceConfiguration);
+
+      rootElement.appendChild(serverSideConfigurationElem);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void validateParametersForTranslationToServiceConfig(Document doc
+    , ServiceCreationConfiguration<ClusteringService> serviceCreationConfiguration) {
+    Objects.requireNonNull(doc, "Document must not be NULL");
+    Objects.requireNonNull(serviceCreationConfiguration, "ServiceCreationConfiguration must not be NULL");
+    if (!ClusteringServiceConfiguration.class.isInstance(serviceCreationConfiguration)) {
+      throw new IllegalArgumentException("Parameter serviceCreationConfiguration must be cast to ClusteringServiceConfiguration."
+                                         + "Provided type of parameter is : " + serviceCreationConfiguration.getClass());
+    }
+  }
+
+  private Element createRootUrlElement(Document doc, ClusteringServiceConfiguration clusteringServiceConfiguration) {
+    Element rootElement = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + CLUSTER_ELEMENT_NAME);
+    rootElement.setAttributeNS(XMLConstants.XMLNS_ATTRIBUTE_NS_URI, "xmlns:" + TC_CLUSTERED_NAMESPACE_PREFIX, NAMESPACE.toString());
+    Element urlElement = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + CONNECTION_ELEMENT_NAME);
+    urlElement.setAttribute(URL_ATTRIBUTE_NAME, clusteringServiceConfiguration.getClusterUri().toString());
+    rootElement.appendChild(urlElement);
+    doc.appendChild(rootElement);
+    return rootElement;
+  }
+
+  private Element createServerElement(Document doc, ClusteringServiceConfiguration clusteringServiceConfiguration) {
+    if (!ConnectionSource.ServerList.class.isInstance(clusteringServiceConfiguration.getConnectionSource())) {
+      throw new IllegalArgumentException("When connection URL is null, source of connection MUST be of type ConnectionSource.ServerList.class");
+    }
+    ConnectionSource.ServerList servers = (ConnectionSource.ServerList)clusteringServiceConfiguration.getConnectionSource();
+    if (servers.getServers() == null || !servers.getServers().iterator().hasNext()) {
```

Comment:
I don't think there is a good reason for ServerList.getServers to return an `Iterable`. So can switch it to Collection or Set

Reply:
ServerList.gerServers was introduced as part of IPv6 support. Platform api takes an Iterable of InetSocketAddress as an input for creating connection and that's why an Iterable is kept inside ServerList.